Jetpack Settings: Add Spam filtering settings card for Jetpack sites to Security tab

oskosk opened the pull request:
This PR introduces a new card, under the Security tab for configuring Spam filtering (Akismet) API key.

Fixes #12195 

##### Collapsed

![image](https://cloud.githubusercontent.com/assets/746152/25857757/6468a204-34b0-11e7-8ef1-8b1533dd74e1.png)

![image](https://cloud.githubusercontent.com/assets/746152/25857784/743ec514-34b0-11e7-8313-e7181efa9804.png)


#### Expanded with an empty input

![image](https://cloud.githubusercontent.com/assets/746152/25856649/020bd818-34ad-11e7-929e-6d0e793a5490.png)



#### Expanded after saving empty input

![image](https://cloud.githubusercontent.com/assets/746152/25857733/52d45de4-34b0-11e7-94bc-c2d34f5249a9.png)



#### After trying to save wrong key

![image](https://cloud.githubusercontent.com/assets/746152/25856576/c34ce978-34ac-11e7-8ea0-1fe6337f13f4.png)


#### Expanded with a valid key

![image](https://cloud.githubusercontent.com/assets/746152/25856549/b4364254-34ac-11e7-9820-5e84015356b5.png)

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings/index.jsx:
```diff
@@ -0,0 +1,82 @@
+							checked={ fields.akismet_show_user_comments_approved || false }
+							disabled={ isRequestingSettings || isSavingSettings || ! akismetActive }
+							onChange={ handleAutosavingToggle( 'akismet_show_user_comments_approved' ) } >
+							{ translate( 'Show the number of approved comments beside each comment author.' ) }
```
Hi! I've found a possible matching string that has already been translated **37** times:
`translate( 'Show the number of approved comments beside each comment author' )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings/index.jsx:
```diff
@@ -0,0 +1,89 @@
+							</InfoPopover>
+						</div>
+					<FormFieldset>
+						<FormLabel>{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **39** times:
`translate( 'Get your API key' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,111 @@
+							</InfoPopover>
+						</div>
+					<FormFieldset>
+						<FormLabel>{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

rickybanister commented:
Here's an image of the latest wp-admin version:

![image](https://cloud.githubusercontent.com/assets/437258/24726281/d2151e1e-1a1f-11e7-94ef-d9bf800e51b5.png)

The bottom padding is a bit off in wp-admin, but otherwise looks right. 

In terms of this PR, it would be nice to align the info icon like the wp-admin screenshot, and clearly to actually show the API key with validation.

rickybanister commented:
Hey @oskosk I think we're pretty close to finished with this one, can you provide an update on where we're at? I think we just needed to address the few small visual items mentioned in my last comment.

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,126 @@
+						</InfoPopover>
+					</div>
+					<FormFieldset>
+						<FormLabel>{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,121 @@
+						</InfoPopover>
+					</div>
+					<div className="site-settings__child-settings">
+						<FormLabel>{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

tyxla replied:
I think it sounds reasonable to use this suggestion instead of "Your API key".

oskosk replied:
I'd try to keep this simillar to the Jetpack Admin Card which reads this same text

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+	onChangeField,
+	translate
+} ) => {
+	const { wordpress_api_key, akismet: akismetActive } = fields;
```
I think that would be more readable if we keep every property on a separate line.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+				<FormFieldset>
+					<div className="spam-filtering-settings__info-link-container site-settings__info-link-container">
+						<InfoPopover >
+							<ExternalLink target="_blank" icon
```
Feels a little odd to keep some props on the same line, while others are separated below. Let's be consistent with this.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+					<div className="spam-filtering-settings__info-link-container site-settings__info-link-container">
+						<InfoPopover >
+							<ExternalLink target="_blank" icon
+								href={ 'https://jetpack.com/features/security/spam-filtering/' }>
```
Nitpicking, but I prefer closing the `>` on a separate line, with one tab to the left. This way you can see where the opening tag ends. Otherwise the eye might think that the `{ translate...` is part of a prop you're passing there.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+						</InfoPopover>
+					</div>
+					<div className="site-settings__child-settings">
+						<FormLabel>{ translate( 'Your API Key ' ) }</FormLabel>
```
This will need a `htmlFor` so it can focus the field when clicked.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+								<FormSettingExplanation>
+									{ translate(
+										'If you don\'t already have an API key, then' +
+										' {{ExternalLink}}Get your API key here{{/ExternalLink}},' +
```
`Get` should probably be lowercase as it's in the middle of a sentence.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+										' and you\'ll be guided through the process of getting one in a new window.',
+										{
+											components: {
+												ExternalLink: (
```
Maybe we could just name this component `link` for simplicity.

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+			'is-error': wordpress_api_key && hasAkismetKeyError
+		} );
+	return (
+		<div className="site-settings__spam-filtering">
```
It feels like there is something wrong with the spacing here, see the screenshot:

![](https://cldup.com/RB4r-6kRPx.png)

Couple of things here:
* The "Your site is protected from spam" header feels a little off, compared to the rest.
* The margin below the card is less than the margin below the rest of the cards.
* The left/right margin next to the API key label and field are not consistent with anything else in settings, including the bruteforce card, which should be similar in terms of layout.

/cc @rickybanister for design feedback on that one.

oskosk replied:
> The "Your site is protected from spam" header feels a little off, compared to the rest.

Indeed!

> The margin below the card is less than the margin below the rest of the cards.

I can't say that I'm perceiving the same. Looks the same amount of space

> The left/right margin next to the API key label and field are not consistent with anything else in settings, including the bruteforce card, which should be similar in terms of layout.

The margin here is consistent with the cards that provide child settings, that's the class that's being applied here

tyxla replied:
> I can't say that I'm perceiving the same. Looks like the same amount of space

It is visible only when you expand the card. When collapsed, spacing is fine.

oskosk replied:
I've checked this with the card expanded:

![image](https://cloud.githubusercontent.com/assets/746152/25848204/a428c778-348f-11e7-9035-52b552156810.png)

tyxla replied:
I am talking about the space below the card itself. Not below the card contents. You can clearly see the difference in my screenshot. Compare the space above your card with the space below your card and you'll see what I'm talking about.

oskosk replied:
oh right!

tyxla replied:
Sorry if I deceived you initially.

oskosk replied:
This is fixed now. The left/margin observation you made still applies, but it's due to the children settings class we use here and there.

tyxla replied:
> This is fixed now. 

Thanks for addressing that 👍 

> The left/margin observation you made still applies, but it's due to the children settings class we use here and there.

Actually it's inconsistent with the rest of the spacing we use generally, but it's the same case like the Protect card. To get around this, you should simply add the `site-settings__foldable-card` class like we do in the Protect card. No further modifications will be necessary.

oskosk replied:
I thinks this was done already.

oskosk commented on client/state/jetpack/settings/utils.js:
```diff
@@ -48,6 +48,9 @@ export const normalizeSettings = ( settings ) => {
 	return Object.keys( settings ).reduce( ( memo, key ) => {
 		switch ( key ) {
+			// Jetpack's settings endpoint in version 4.9 does not support receiving 'akismet' among the settings
+			case 'akismet':
```
AFAIK the API has never accepted this option name as valid.

tyxla replied:
Maybe we can combine all settings that we ignore together above a single `break`?

oskosk commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,122 @@
+} ) => {
+	const { wordpress_api_key, akismet: akismetActive } = fields;
+	const isDirty = includes( dirtyFields, 'wordpress_api_key' );
+	const header = ( akismetActive && wordpress_api_key && ! isDirty )
```
> character in the API key field, then delete it to return to the previous state, it will not display that it's valid, unless you refresh.

I think this is fine. It represents the logic of a field being dirty until you save it. 

> The same happens if you have an empty field or a wrong API key, and then you submit with the valid API key - you will still see the "Your site needs an Antispam key." message, and the field validation will not be

I'll check this, this is wrong. Thank you

tyxla replied:
> I think this is fine. It represents the logic of a field being dirty until you save it.

Nope, IMO it's not. A field is supposed to be dirty only if it's value differs from the last one we have. Which is not the case if you've returned the field to the original value.

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,129 @@
+						</InfoPopover>
+					</div>
+					<div className="site-settings__child-settings">
+						<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,127 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

oskosk commented:
Thanks @tyxla. I've addressed most of the observations. I'll ping you when I'm done with what's left. The ones left are:

1. Fix the behaviour of the validation message when the input value is the same as the value loaded from the DB (which we assume is always valid)
1. Improve the error message for notices. _maybe_ :trollface:

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,131 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,131 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,130 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key ' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,130 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,131 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

rickybanister commented:
The spacing looks strange (I think) because in Calypso (but not wp-admin) we are indenting the API key field. Also we do not have the checkmark next to the 'your site is protected from spam'. The checkmark would indent the status message to align with the api key field. 

Also if you are not protected, we should be showing a red alert icon. 

Otherwise this looks close.

oskosk commented:
>Also we do not have the checkmark next to the 'your site is protected from spam'. The checkmark would indent the status message to align with the api key field.
> Also if you are not protected, we should be showing a red alert icon.

OK, will add them!.

tyxla commented:
@oskosk let me know when this one is ready for another spin.

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,142 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

oskosk commented:
@tyxla I missed your last comment :(. It's ready for another review, thanks

tyxla commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,142 @@
+};
+
+SpamFilteringSettings.propTypes = {
+	hasAkismetKeyError: PropTypes.bool,
```
Nit: Jon would alphabetize these 😆

tyxla commented on client/my-sites/site-settings/style.scss:
```diff
@@ -339,6 +339,12 @@
 			color: $alert-green;
 		}
+
+		.gridicons-notice-outline {
```
Since these styles are the same for `.gridicons-checkmark`, can we combine them and just keep the `color` separate?

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,143 @@
+					</InfoPopover>
+				</div>
+				<div className="site-settings__child-settings">
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,143 @@
+							</ExternalLink>
+						</InfoPopover>
+					</div>
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

oskosk commented:
@tyxla I've addressed your observations, but have a few comments.


> Even when I currently have a valid key set, and I just click Save, while saving, I see the "Your site needs an Antispam key" message, which is not right:

I've just updated this to not show any text while loading or saving. 

> While the key data is loading, I see the "Your site needs an Antispam key." message. Then, after data loads, it properly says "Your site is protected from spam". This feels wrong, can we replace the falsy error message with a placeholder that's shown until the data we need loads?

So, yes, but this is the same non-placeholder approach we followed on all of the settings. As when toggles show as deactivated when the data is not really there, and they switch to `on` when data arrives. 

But removing the message at all while saving and loading this looks better now. This problem emerged because we added this style of messages asserting truths while in fact is something doubtful. The same goes for the stats settings card which asserts you're collecting stats right away. 

> When I have a currently valid key set, and type a new character to make it invalid, I immediately see the "Your site needs an Antispam key." message. This is wrong - I haven't saved it yet, so my key is still valid. Doesn't it make sense to have an error message inline (FormInputValidation)?

I wouldn't say it's wrong because you actually still a need antispam key, which you're currently updating from a valid one to something not valid which will only be checked by saving. But, removing the message while the field is dirty also makes this look better. The FormInputValidation is there for showing when you have just saved a wrong/right key. I chose to hide it while you're typing (well, actually while you have changed the value of the input and it does not match the loaded valid key). 


> The info icon appears a little more to the top than it does for the rest of the cards. Is that on purpose? See the SSO card for comparison:

No, just fixed that. I didn't realise we were using component-specific styles for the protect card that were defined in the general settings styles file. This seems to be one of the few cases where the InfoPopover is placed inside the child settings div. 

Can you give it another check please?

Regarding the behaviour on validation, I don't think we should be giving it more iterations, because it's already working different than in the Jetpack side, which, in addition, has validation on key change. Now, the thing is that it is the only place where we actually do validation on change. I'd prefer to merge this for once and tackle updates or bugs in new PRs.

tyxla commented:
Thanks for your patience and diligence with this one, Osk!

Although I really don't want to block this one anymore, there are several things that I don't feel are right:

* Right after data loads, also when you add a character to the field to invalidate it, there is some small jumping which looks a little broken. This can be fixed by adding some height to the empty box.
* The empty foldable card header during loading looks odd. Can we add a placeholder there instead? You can use the `placeholder()` sass function for that. Or if you don't like that idea, maybe we can hide the entire foldable card?
* If you input an invalid key, the header becomes empty again; which feels odd. It is the same case if you don't have a key - you simply end up with an empty header and empty field when you expand the foldable card. @rickybanister what can we do to make this one less odd?
* If I save the Brute Force card, the Antispam card header becomes empty. Why would it even change, I'm even not interacting with that card?

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -0,0 +1,134 @@
+							</ExternalLink>
+						</InfoPopover>
+					</div>
+					<FormLabel htmlFor="wordpress_api_key">{ translate( 'Your API Key' ) }</FormLabel>
```
Hi! I've found a possible matching string that has already been translated **42** times:
`translate( 'API Key' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/akismet-card-for-jetpack-sites-in-settings-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

rickybanister commented:
>>>If you input an invalid key, the header becomes empty again; which feels odd. It is the same case if you don't have a key - you simply end up with an empty header and empty field when you expand the foldable card. @rickybanister what can we do to make this one less odd?

Can you help me understand a little better @tyxla? Can I see a picture of that? 

I agree with you that the header text should only change once the user clicks the save button. At that point the header can be updated to reflect the current status of the feature. I think the important thing to remember is that the header should display the status of spam prevention on their site, not necessarily other meta statuses. So we could say something more like `(!) Your site is vulnerable from spam because your API key is invalid`. That would ground the card in what's really happening. 

I can't get calypso.live to work for me right now so I can't test this properly at the moment. Let me know if my comment makes any sense of it I'm just wasting time.

tyxla commented:
> Can you help me understand a little better @tyxla? Can I see a picture of that?

Sure thing, @rickybanister. I'll provide steps in case you can get calypso.live to work, but also a screenshot so you can see what I'm talking about if calypso.live doesn't work for you:

* Load the Security tab for a site that has Akismet configured correctly.
* Open the Antispam card.
* The key is currently valid, and you see a "Your site is protected from spam." message.
* Type a character to invalidate the key.
* You will see a blank card header, even though you haven't saved yet:

![](https://cldup.com/0UrsJtPRmP.png)

Let me know if that helps, or if you need further instructions and/or screenshots.

rickybanister commented:
@tyxla sorry for all the back and forth.

I think if we have no key we can say `Protect your site from spam by adding an Akismet key` which is slightly different than what has been proposed, but it contextualizes the error around the larger goal of the feature.

Does that work?

If they haven't saved the field yet, it's still 100% fine to continue showing that message until we can validate the key they enter.

tyxla commented:
Thanks @rickybanister, I think that'd work just fine. And I really don't want to block this PR anymore, especially if you're fine with the design part.

rickybanister commented:
Yup, fine with it at this point, let's get this merged!
